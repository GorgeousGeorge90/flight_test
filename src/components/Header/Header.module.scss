


.header_container {
  @apply flex items-center justify-center bg-blue-600 text-zinc-100;
  height: 200px;

  .header_content {
    @apply p-14 w-full;

    .header_logo {
      @apply flex items-center gap-10;

      .header_info {
        @apply font-serif flex-grow;

        .header_info_title {
          @apply text-2xl;
        }

        .header_info_text {
          @apply first-letter:uppercase;
        }
      }

      .header_user_info {
        @apply flex flex-row-reverse gap-4 justify-center;

        .header_user_about {
          @apply flex flex-col items-center gap-1;

          .btn {
            @apply first-letter:uppercase bg-violet-500 rounded-md p-1 hover:bg-orange-500;
          }
        }

        .header_user_list {
          @apply flex flex-row-reverse gap-4;
        }
      }

      .header_user_ava {
        height: 60px;
      }
    }
  }
}