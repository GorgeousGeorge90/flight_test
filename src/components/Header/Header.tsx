import styles from './Header.module.scss';


const Header = () => {


    return (<header className={styles.header_container}>
                <div className={styles.header_content}>
                    <div className={styles.header_logo}>

                    </div>
                    <div className={styles.header_title}>
                        Header
                    </div>
                </div>
            </header>)
}

export default Header