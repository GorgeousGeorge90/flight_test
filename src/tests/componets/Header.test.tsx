import Header from '../../components/Header/Header';
import { render } from '@testing-library/react';


describe('Header',()=> {
    it('goal:render Header component', ()=> {
        const view = render(<Header/>)
        expect(view).toMatchSnapshot()
    })
})


