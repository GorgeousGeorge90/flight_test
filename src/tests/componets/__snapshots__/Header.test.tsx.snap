// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header goal:render Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header_container"
      >
        <div
          class="header_content"
        >
          <section
            class="header_logo"
          >
            <img
              alt="logo"
              src="logo.png"
            />
            <section
              class="header_info"
            >
              <h1
                class="header_info_title"
              >
                Northern Hermes
              </h1>
              <p
                class="header_info_text"
              >
                only good offers!
              </p>
            </section>
            <section
              class="header_input"
            >
              <section
                class="user_form_container"
              >
                <form
                  class="user_form_content"
                >
                  <input
                    class="user_form_input"
                    name="name"
                    placeholder="Введите имя"
                    type="text"
                  />
                  <button
                    class="user_form_button"
                  >
                    авторизоваться
                  </button>
                </form>
              </section>
            </section>
            <section
              class="header_user_info"
            >
              <p>
                not authorized
              </p>
            </section>
          </section>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header_container"
    >
      <div
        class="header_content"
      >
        <section
          class="header_logo"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <section
            class="header_info"
          >
            <h1
              class="header_info_title"
            >
              Northern Hermes
            </h1>
            <p
              class="header_info_text"
            >
              only good offers!
            </p>
          </section>
          <section
            class="header_input"
          >
            <section
              class="user_form_container"
            >
              <form
                class="user_form_content"
              >
                <input
                  class="user_form_input"
                  name="name"
                  placeholder="Введите имя"
                  type="text"
                />
                <button
                  class="user_form_button"
                >
                  авторизоваться
                </button>
              </form>
            </section>
          </section>
          <section
            class="header_user_info"
          >
            <p>
              not authorized
            </p>
          </section>
        </section>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
